File: Flatshare Code/FlatShare 2.0/FlatshareCentral/src/main/java/com/joinflatshare/customviews/bottomsheet/BottomSheetView.java
package com.joinflatshare.customviews.bottomsheet;

import android.app.Activity;
import android.os.Handler;
import android.os.Looper;
import android.view.View;

import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.bottomsheet.BottomSheetBehavior;
import com.joinflatshare.FlatshareCentral.R;
import com.joinflatshare.interfaces.OnitemClick;

import java.util.ArrayList;

public class BottomSheetView {
    private Activity activity;
    private ArrayList<ModelBottomSheet> modelBottomSheets;
    private BottomSheetAdapter adapter;
    private BottomSheetBehavior<ConstraintLayout> bottomSheetBehavior;

    public BottomSheetView(Activity activity, ArrayList<ModelBottomSheet> modelBottomSheets) {
        this.activity = activity;
        this.modelBottomSheets = modelBottomSheets;
        init();
    }

    private void init() {
        bottomSheetBehavior = BottomSheetBehavior.from(activity.findViewById(R.id.bottomSheet));
        RecyclerView rv = activity.findViewById(R.id.rv_bottomsheet);
        rv.setLayoutManager(new LinearLayoutManager(activity));
        adapter = new BottomSheetAdapter(activity, modelBottomSheets);
        rv.setAdapter(adapter);
        bottomSheetBehavior.setFitToContents(true);
        bottomSheetBehavior.setHideable(true);
        bottomSheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
    }

    public void show(OnitemClick onitemClick) {
        activity.findViewById(R.id.bottomSheet).setVisibility(View.VISIBLE);
        new Handler(Looper.getMainLooper()).postDelayed(() -> bottomSheetBehavior.setState(BottomSheetBehavior.STATE_EXPANDED),500);
        adapter.setClickListener((view, position) -> {
            onitemClick.onitemclick(view, position);
            bottomSheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
        });
    }
}

File: Flatshare Code/FlatShare 2.0/FlatshareCentral/src/main/java/com/joinflatshare/customviews/bottomsheet/BottomSheetView.kt
package com.joinflatshare.customviews.bottomsheet

import android.view.View
import androidx.activity.ComponentActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.android.material.bottomsheet.BottomSheetDialog
import com.joinflatshare.FlatshareCentral.databinding.DialogBottomsheetBinding
import com.joinflatshare.interfaces.OnitemClick
import com.joinflatshare.ui.bottomsheet.BottomSheetBaseView

/**
 * Created by debopam on 09/07/24
 */
class BottomSheetView(
    private val activity: ComponentActivity,
    private val modelBottomSheets: ArrayList<ModelBottomSheet>,
    private val onItemClick: OnitemClick
) : BottomSheetBaseView(activity) {
    private lateinit var viewBind: DialogBottomsheetBinding
    private lateinit var adapter: BottomSheetAdapter
    private lateinit var dialog: BottomSheetDialog

    init {
        create()
    }

    private fun create() {
        dialog = BottomSheetDialog(activity)
        viewBind = DialogBottomsheetBinding.inflate(activity.layoutInflater)
        dialog.setContentView(viewBind.root)
        setup()
        click()
        showDialog(dialog)
    }

    private fun setup() {
        viewBind.rvBottomsheet.layoutManager = LinearLayoutManager(activity)
        adapter = BottomSheetAdapter(activity, modelBottomSheets)
        viewBind.rvBottomsheet.adapter = adapter
    }

    private fun click() {
        adapter.setClickListener { view: View?, position: Int ->
            onItemClick.onitemclick(view, position)
            dismissDialog(dialog)
        }
    }
}